sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/MetricDataMapper.java

```diff
@@ -149,9 +151,17 @@ public static void updateMetricPointBuilder(
         }
 
         pointBuilder.setValue(pointDataValue);
-        // TODO (heya) why give it the same name as otel metric?
-        //  it seems this field doesn't matter and only _MS.MetricId property matters?
-        pointBuilder.setName(metricData.getName());
+
+        // We emit some metrics via OpenTelemetry that have names which use characters that aren't
+        // supported in OpenTelemetry metric names, and so we put the real metric names into an attribute
+        // (where these characters are supported) and then pull the name back out when sending it to Breeze.
+        String metricName = pointData.getAttributes().get(APPLICATIONINSIGHTS_INTERNAL_METRIC_NAME);
```

Comment:
I think we also want to skip this attribute when emitting it to Breeze

Reply:
updated the comment